Tests/FluentAssertions.Specs/Numeric/NumericDifferenceAssertionsSpecs.cs

```diff
@@ -0,0 +1,703 @@
+using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Numeric
+{
+    public class NumericDifferenceAssertionsSpecs
+    {
+        public class Be
+        {
+            [Theory]
+            [InlineData(8, 5)]
+            [InlineData(1, 9)]
+            public void When_an_int_difference_between_small_numbers_it_should_not_add_difference_message(int value, int expected)
+            {
+                // Act
+                Action act = () =>
+                    value.Should().Be(expected, "because we want to test the failure {0}", "message");
+
+                // Assert
+                act
+                    .Should().Throw<XunitException>()
+                    .WithMessage($"Expected value to be {expected} because we want to test the failure message, but found {value}.");
+            }
+
+            [Theory]
+            [InlineData(50, 20)]
+            [InlineData(20, 50)]
+            [InlineData(123, -123)]
+            [InlineData(-123, 123)]
+            public void When_an_int_difference_between_large_numbers_it_should_add_difference_message(int value, int expected)
+            {
+                // Act
+                Action act = () =>
+                    value.Should().Be(expected, "because we want to test the failure {0}", "message");
+
+                // Assert
+                act
+                    .Should().Throw<XunitException>()
+                    .WithMessage($"Expected value to be {expected} because we want to test the failure message, but found {value} (difference of {value - expected}).");
```

Comment:
🤔 Thinking, should we add the expected difference to the `InlineData` instead of calculating it for the assertion?